### ISSUE TITLE
Refactor tree implementation to more closely match table

### ISSUE THREAD

**Issue by mikerodonnell89** [SAP/fundamental-ngx#270]
#### Is this a bug, enhancement, or feature request?

Enhancement

#### Briefly describe your proposal.

Recently we refactored the table component to give the developer more control over the contents of the table.  The current tree implementation still reflects the more opinionated previous version of the table.  We should refactor the tree so using the component more closely resembles the new table implementation.
What is needed for applications like smartedit - drag and drop of rows

#### Which versions of Angular and Fundamental NGX are affected? (If this is a feature request, use current version.)

Angular 6, fundamental-ngx 0.0.20-beta

**Comment by MattL75** [comment]
Blocked: currently waiting on new information from the clouds

**Comment by hisabimbola** [comment]
Hello guys,

any update regarding this?

I would like to implemented something like this [here](https://sapui5.hana.ondemand.com/#/entity/sap.ui.table.TreeTable/sample/sap.ui.table.sample.TreeTable.JSONTreeBinding) and I think this issue may help with that

**Comment by MattL75** [comment]
Hey, we have no further updates. I started preliminary design for an API around 4-5 months ago, but I was told to put it aside until further notice. Haven't touched it since then. If I recall correctly, there was a redesign coming up, so @droshev told me to hold off. I don't think the redesign would be massive enough that it would mean changing the API though.

**Comment by hisabimbola** [comment]
So can we now use the current implementation and also it's not present in the documentation. Is this also part of the plan?

**Comment by mikerodonnell89** [comment]
@droshev can probably close this, since many changes are coming to table, I imagine this is no longer relevant
